blackpill-f4: #define PLATFORM_HAS_POWER_SWITCH leads to error: implicit declaration of function 'adc_start_conversion_direct'

Using 9fe5889, adding ```#define PLATFORM_HAS_POWER_SWITCH``` to blackpill-f4.h and compiling using the command ```make PROBE_HOST=blackpill-f401cc``` leads to the following error.

```vim
  CC      platforms/common/stm32/timing_stm32.c
platforms/common/stm32/timing_stm32.c: In function 'sys_tick_handler':
platforms/common/stm32/timing_stm32.c:110:25: error: implicit declaration of function 'adc_start_conversion_direct'; did you mean 'adc_start_conversion_injected'? [-Werror=implicit-function-declaration]
  110 |                         adc_start_conversion_direct(ADC1);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                         adc_start_conversion_injected
cc1: all warnings being treated as errors
make[1]: *** [timing_stm32.o] Error 1
make: *** [all] Error 2
```

This error was not present a couple months ago when compiling in the same way.

--- a/src/platforms/common/stm32/timing_stm32.c
+++ b/src/platforms/common/stm32/timing_stm32.c
@@ -33,7 +33,7 @@ static volatile uint32_t time_ms = 0;
 uint32_t target_clk_divider = 0;
 
 static size_t morse_tick = 0;
-#ifdef PLATFORM_HAS_POWER_SWITCH
+#if defined(PLATFORM_HAS_POWER_SWITCH) && defined(STM32F1)
 static uint8_t monitor_ticks = 0;
 
 /* Derived from calculating (1.2V / 3.0V) * 4096 */
@@ -90,7 +90,7 @@ void sys_tick_handler(void)
 	} else
 		++morse_tick;
 
-#ifdef PLATFORM_HAS_POWER_SWITCH
+#if defined(PLATFORM_HAS_POWER_SWITCH) && defined(STM32F1)
 	/* First check if target power is presently enabled */
 	if (platform_target_get_power()) {
 		/*

